src/ssl.c

```diff
@@ -258,6 +260,533 @@ const WOLF_EC_NIST_NAME kNistCurves[] = {
+
+            /* skip over version and length */
+            echConfig += 4;
+
+            /* configId, 1 byte */
+            workingConfig->configId = *(echConfig);
+            echConfig++;
+            /* kemId, 2 bytes */
+            ato16(echConfig, &workingConfig->kemId);
+            echConfig += 2;
+            /* hpke public_key length, 2 bytes */
+            ato16(echConfig, &hpkePubkeyLen);
+            echConfig += 2;
+            /* hpke public_key */
+            XMEMCPY(workingConfig->receiverPubkey, echConfig,
+              hpkePubkeyLen);
+            echConfig += hpkePubkeyLen;
+            /* cipherSuitesLen */
+            ato16(echConfig, &cipherSuitesLen);
+            workingConfig->cipherSuites =
+                (EchCipherSuite*)XMALLOC(cipherSuitesLen,
+                ssl->heap, DYNAMIC_TYPE_TMP_BUFFER);
+            echConfig += 2;
+            workingConfig->numCipherSuites = cipherSuitesLen / 4;
+            /* cipherSuites */
+            for (j = 0; j < workingConfig->numCipherSuites; j++) {
+                ato16(echConfig + j * 4, &workingConfig->cipherSuites[j].kdfId);
+                ato16(echConfig + j * 4 + 2,
+                    &workingConfig->cipherSuites[j].aeadId);
+            }
+            echConfig += cipherSuitesLen;
+            /* publicNameLen */
+            ato16(echConfig, &publicNameLen);
+            workingConfig->publicName = (char*)XMALLOC(publicNameLen + 1,
+                ssl->heap, DYNAMIC_TYPE_TMP_BUFFER);
+            echConfig += 2;
+            /* publicName */
+            XMEMCPY(workingConfig->publicName, echConfig, publicNameLen);
+            /* null terminated */
+            workingConfig->publicName[publicNameLen] = 0;
+
+            /* add length to go to next config, +4 for version and length */
+            i += length + 4;
+
+            /* check that we support this config */
+            for (j = 0; j < HPKE_SUPPORTED_KEM_LEN; j++) {
+                if (hpkeSupportedKem[j] == workingConfig->kemId)
+                    break;
+            }
+
+            /* if we don't support the kem or at least one cipher suite */
+            if (j >= HPKE_SUPPORTED_KEM_LEN ||
+              EchConfigGetSupportedCipherSuite(workingConfig) < 0) {
+                XFREE(workingConfig->cipherSuites, ssl->heap,
+                    DYNAMIC_TYPE_TMP_BUFFER);
+                XFREE(workingConfig->publicName, ssl->heap,
+                    DYNAMIC_TYPE_TMP_BUFFER);
+                XFREE(workingConfig->raw, ssl->heap, DYNAMIC_TYPE_TMP_BUFFER);
+                workingConfig = lastConfig;
+            }
+        } while ((word32)i < echConfigsLen);
+    }
+
+    /* if we found valid configs */
+    if ( configList != NULL ) {
+        ssl->options.useEch = 1;
+        ssl->echConfigs = configList;
+
+        return WOLFSSL_SUCCESS;
+    }
+
+    return WOLFSSL_FATAL_ERROR;
+}
+
+int wolfSSL_GetEchConfig(WOLFSSL_EchConfig* config, byte* output,
```

Comment:
Change name to something not suggesting it should take a WOLFSSL object.
ws_GetEchConfig or EchConfig_Get